### ISSUE TITLE
git-sv release-notes fails to find tag in pipeline

### ISSUE THREAD

**Issue by JashanChopra** [bvieira/sv4git#84]
Hello, loving the tool, but running into an issue with the release-notes tool that I cannot figure out.

In my pipeline:

```
$ git fetch --tags
$ git tag
1.0.0
1.1.0
1.1.1
1.1.2
$ git sv release-notes -t "$CI_COMMIT_TAG"
ERROR: error listing tags, message: tag: 1.1.2 not found
```

Locally it works as expected, so it's likely an issue somewhere in my pipeline configuration, but the error message isn't giving me many clues. Any idea why sv4git can't seem to find the tag even though it is being listed as an output of `git tag`?

**Comment by JashanChopra** [comment]
Looks like it is a config issue. 

It seems that your config merging doesn't respect when the tag.filter is set to `''`, which we have. Thus it is using a bad default filter we set upstream as the default. 

Seems the issue exists in the tool you are using for merging configs: https://github.com/imdario/mergo/issues/190

**Comment by bvieira** [comment]
hey, thanks for using sv4git!! ❤️

I had some problems with my environment and need to install everything again.. probably I'll take a look on this weekend..

the [tag.filter](https://github.com/bvieira/sv4git/blob/e76d0b8c1a90bc34a6c051287dbc00bcfdeefede/cmd/git-sv/config.go#L82) is empty by default, are you overwriting it with some value?

can you send me your config, both the repo and user?

the problem is that in user config the filter have some values and the repo config is setting it as an empty string?

**Comment by JashanChopra** [comment]
We had a user level config (set as the variable) overwriting the default filter.

```
version: "1.1" 

tag:
    pattern: '%d.%d.%d' # Pattern used to create git tag.
    filter: 'v?[0-9]'

[rest is default]
```

But at the repository level it was an empty filter:

```
version: "1.1" 

tag:
    pattern: 'v?[0-9].[0-9].[0-9]' # Pattern used to create git tag.
    filter: ''

[rest is default]
```

Expected behavior is that the repository level filter will overwrite the user level filter, however, in the case that the repository level filter is an empty string, it does not successfully overwrite. Which looks like an issue originating from a downstream dependency you use: https://github.com/imdario/mergo/issues/190.

Running `git-sv cfg show` in our pipeline makes it clear that the repository level config is getting precedent except for filter:

```
version: "1.1" 

tag:
    pattern: 'v?[0-9].[0-9].[0-9]' # Pattern used to create git tag.
    filter: 'v?[0-9]'
```

**Comment by bvieira** [comment]
ohh.. great.. now i understood..

thanks a lot for let me know.. I'll work on this

**Comment by bvieira** [comment]
hey, this feature is available on [v2.9.0](https://github.com/bvieira/sv4git/releases/tag/v2.9.0) :rocket: !!

Thanks again and just let me know if everything is working as expected!!
